src/parser/name_resolution.c

```diff
@@ -7087,76 +7114,73 @@ pt_resolve_using_index (PARSER_CONTEXT * parser, PT_NODE * index, PT_NODE * from
 	}
 
       /* the specified class in "class.index" does not exist in spec list */
-      PT_ERRORmf (parser, index, MSGCAT_SET_PARSER_SEMANTIC, MSGCAT_SEMANTIC_USING_INDEX_ERR_2,
-		  pt_short_print (parser, index));
+      pt_write_warning (parser, index, __LINE__, MSGCAT_SET_PARSER_SEMANTIC, MSGCAT_SEMANTIC_USING_INDEX_ERR_2);
       return NULL;
     }
-  else
-    {				/* if (index->info.name.resolved != NULL) */
-      /* index name without class name specification */
 
-      /* find the class of the index from spec list */
-      for (spec = from; spec; spec = spec->next)
+  /* case (index->info.name.resolved == NULL) */
+  /* index name without class name specification */
+
+  /* find the class of the index from spec list */
+  for (spec = from; spec; spec = spec->next)
+    {
+      if (spec->node_type != PT_SPEC)
 	{
-	  if (spec->node_type != PT_SPEC)
+	  PT_INTERNAL_ERROR (parser, "resolution");
+	  *is_ignore = false;
+	  return NULL;
+	}
+
+      range = spec->info.spec.range_var;
+      entity = spec->info.spec.entity_name;
+      if (range != NULL && entity != NULL && entity->info.name.original != NULL)
+	{
+	  classop = db_find_class (entity->info.name.original);
+	  if (classop == NULL)
 	    {
-	      PT_INTERNAL_ERROR (parser, "resolution");
-	      return NULL;
+	      continue;
 	    }
-
-	  range = spec->info.spec.range_var;
-	  entity = spec->info.spec.entity_name;
-	  if (range != NULL && entity != NULL && entity->info.name.original != NULL)
+	  if (au_fetch_class (classop, &class_, AU_FETCH_READ, AU_SELECT) != NO_ERROR)
```

Comment:
Shouldn't the error also be printed as pt_write_warning?

Reply:
Changed only for MSGCAT_SET_PARSER_SEMANTIC of the existing code.